dev/TabView/TabViewItem.cpp

```diff
@@ -106,12 +120,81 @@ winrt::AutomationPeer TabViewItem::OnCreateAutomationPeer()
     return winrt::make<TabViewItemAutomationPeer>(*this);
 }
 
+void TabViewItem::OnCloseButtonOverlayModeChanged(winrt::TabViewCloseButtonOverlayMode const& mode)
+{
+    m_closeButtonOverlayMode = mode;
+    if (mode == winrt::TabViewCloseButtonOverlayMode::OnHover && !IsSelected())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"CloseButtonCollapsed",false);
+    }
+    if (mode == winrt::TabViewCloseButtonOverlayMode::Persistent || mode == winrt::TabViewCloseButtonOverlayMode::Auto)
+    {
+        winrt::VisualStateManager::GoToState(*this, L"CloseButtonVisible", false);
+    }
+}
+
+void TabViewItem::OnTabViewWidthModeChanged(winrt::TabViewWidthMode const& mode)
+{
+    m_tabViewWidthMode = mode;
+    UpdateWidthModeVisualState();
+}
+
 
 void TabViewItem::UpdateCloseButton()
 {
-    if (auto && closeButton = m_closeButton.get())
+    if (!IsClosable())
+    {
+        winrt::VisualStateManager::GoToState(*this, L"CloseButtonCollapsed", false);
+    }
+    else
+    {
+        switch (m_closeButtonOverlayMode)
+        {
+            case winrt::TabViewCloseButtonOverlayMode::Persistent:
+            {
+                // In case of "Persistent" always show button
+                winrt::VisualStateManager::GoToState(*this, L"CloseButtonVisible", false);
+                break;
+            }
+            case winrt::TabViewCloseButtonOverlayMode::Auto:
+            {
+                // In case of "Auto" always show button
+                winrt::VisualStateManager::GoToState(*this, L"CloseButtonVisible", false);
+                break;
+            }
```

Comment:
Since these are handled in the default case, can we just remove them?

Reply:
Yes, makes more sense than having the same code thrice. Thanks for pointing that out!